deepchem/data/data_loader.py

```diff
@@ -1710,41 +1710,56 @@ def _get_shards(self, inputs):
 
         Returns
         -------
-        data
-            list of dictionaries where each dictionary corresponds to one
-            shard and is then featurized into one entry object.
+        current_shard
         """
-        with open(inputs) as f:
-            data = yaml.load(f, Loader=SafeLoader)
-        return (data)
+        data = []
+        for file in inputs:
```

Comment:
This is not incorrect, but it's memory inefficient since it loads the full dataset into memory. We may need to come back and fix later once we move to bigger datasets